pkg/liqoctl/purge/handler.go

```
@@ -0,0 +1,468 @@
+		func(list *netv1alpha1.NetworkConfigList) []*netv1alpha1.NetworkConfig { return pointerList(list.Items) })
+}
+
+func forceDeleteResourceRequests(ctx context.Context, cl client.Client, namespace string, s *pterm.SpinnerPrinter) error {
+	return forceDelete(ctx, cl, namespace, s, &discoveryv1alpha1.ResourceRequest{}, &discoveryv1alpha1.ResourceRequestList{},
+		func(list *discoveryv1alpha1.ResourceRequestList) []*discoveryv1alpha1.ResourceRequest {
+			return pointerList(list.Items)
+		})
+}
+
+func forceDeleteResourceOffers(ctx context.Context, cl client.Client, namespace string, s *pterm.SpinnerPrinter) error {
+	return forceDelete(ctx, cl, namespace, s, &sharingv1alpha1.ResourceOffer{}, &sharingv1alpha1.ResourceOfferList{},
+		func(list *sharingv1alpha1.ResourceOfferList) []*sharingv1alpha1.ResourceOffer {
+			return pointerList(list.Items)
+		})
+}
+
+func forceDeleteNamespaceMaps(ctx context.Context, cl client.Client, namespace string, s *pterm.SpinnerPrinter) error {
+	return forceDelete(ctx, cl, namespace, s, &virtualkubeletv1alpha1.NamespaceMap{}, &virtualkubeletv1alpha1.NamespaceMapList{},
+		func(list *virtualkubeletv1alpha1.NamespaceMapList) []*virtualkubeletv1alpha1.NamespaceMap {
+			return pointerList(list.Items)
+		})
+}
+
+func forceDelete[T client.Object, U client.ObjectList](ctx context.Context, cl client.Client,
+	namespace string, s *pterm.SpinnerPrinter,
+	obj T, objList U, objListToObjArray func(U) []T) error {
+
+	if err := cl.DeleteAllOf(ctx, obj, client.InNamespace(namespace)); err != nil {
+		return err
+	}
+
+	if err := cl.List(ctx, objList, client.InNamespace(namespace)); err != nil {
+		return err
+	}
+
+	for _, obj := range objListToObjArray(objList) {
+		obj.SetFinalizers([]string{}) // nolint:typecheck // golangci is not able to understand it at the moment https://github.com/golangci/golangci-lint/issues/2649
+		if err := client.IgnoreNotFound(cl.Update(ctx, obj)); err != nil {
+			s.WarningPrinter.Printf("%v\n", err)
```

Should probably enrich the error a bit, to make it easier to understand. Should also probably return the error.